File: src/main/java/eotree/EOCopy.java
package eotree;

import java.util.List;
import java.util.stream.Collectors;

public class EOCopy extends EOExpr {
    public EOExpr      trg;
    public List<EOBnd> args;

    public EOCopy(EOExpr trg, List<EOBnd> args) {
        this.trg = trg;
        this.args = args;
    }

    @Override
    public String generateEO(int indent) {
        return trg.generateEO(indent) + args.stream()
                .map(arg -> "\n" + arg.generateEO(indent + 1))
                .collect(Collectors.joining());
    }
}

File: src/main/java/eotree/EOCopy.kt
package eotree

class EOCopy : EOExpr {
    val trg: EODot
    val args: List<EOExpr>

    constructor(trg: String, args: List<EOExpr>) {
        this.trg = trg.eoDot()
        this.args = args
    }

    constructor(trg: EODot, args: List<EOExpr>) {
        this.trg = trg
        this.args = args
    }

    constructor(trg: String, vararg args: EOExpr) : this(trg, args.toList())

    constructor(trg: EODot, vararg args: EOExpr) : this(trg, args.toList())

    override fun generateEO(indent: Int): String {
        return trg.generateEO(indent) + args
            .map { arg -> "\n${arg.generateEO(indent + 1)}" }
            .joinToString("")
    }

    override fun toString(): String =
        "$trg(${args.joinToString()})"
}
